### PR TITLE
Add ToGlibPtrMove trait

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/gtk-rs/gtk-rs-core/issues/50.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Updated.

File: glib/src/variant_type.rs [hunk 1]
```diff
@@ -90,12 +90,14 @@ impl VariantType {
...
-        let ptr = type_string.into_raw();
+        unsafe {
+            let ptr = type_string.into_glib_ptr();
 
-        Ok(VariantType {
-            ptr: ptr as *mut ffi::GVariantType,
-            len,
-        })
+            Ok(VariantType {
+                ptr: ptr as *mut ffi::GVariantType,
```
Comment (sdroege) [inline glib/src/variant_type.rs:97]: Hm why doesn't this use `ptr::NonNull`?

**Review by sdroege — COMMENTED** [review]
Seems good to me

File: glib/src/variant_type.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply glib/src/variant_type.rs:97]: Aren't you the one who wrote this code? :p 

If there is a problem with it, please an open an issue so it's not forgotten.

Comment (sdroege) [reply glib/src/variant_type.rs:97]: I'll update it once this here is merged :) I didn't write it though

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Sending follow-up PRs soon.

**Pull request merged**